graql/gremlin/TraversalPlanFactoryImpl.java

```diff
@@ -219,19 +227,19 @@ public GraqlTraversal createTraversal(Pattern pattern) {
         return weightedGraph;
     }
 
-    private static Set<Node> chooseStartingNodeSet(Set<Fragment> fragmentSet, Map<NodeId, Node> allNodes, SparseWeightedGraph sparseWeightedGraph, Transaction tx) {
+    private Set<Node> chooseStartingNodeSet(Set<Fragment> fragmentSet, Map<NodeId, Node> allNodes, SparseWeightedGraph sparseWeightedGraph) {
         final Set<Node> highPriorityStartingNodeSet = new HashSet<>();
         final Set<Node> lowPriorityStartingNodeSet = new HashSet<>();
```

Comment:
I know it's not a change of this PR again and also this is more personal/stylistic decision, but I think we should avoid using random `final` keywords inside methods. The code inside a method is written by us, these are not structures injected by the user nor that will be shared with the outside world. The methods should never be so long or complex that we lose track of local variables, so that we need to declare them final. My reasoning behind this is: let's not start throwing random defensive mechanisms here and there, and also let's avoid adding unnecessary noise to the whole repo. There are programming languages which have native final variables, such as JS or C++, Java unfortunately doesnt, so we should learn how to and when to use `final` and also what is it really protecting us against? The main problems usually happens when people start mutating collections, it rarely happens that someone inside a method tries to re-allocate a variable. But anyway, food for thoughts 👋